### ISSUE TITLE
Maya-usd (0.22.0) plugin wont load, built with USD Version 22.05b

### ISSUE THREAD

**Issue by jcoelho88** [Autodesk/maya-usd#2909]
**Describe the issue**
I want to test out the suggested fix, it's the instance prim crash issue where current Maya-usd
plugins that are built with USD Version 21.11 have. See ticket here: https://github.com/Autodesk/maya-usd/issues/2721#issuecomment-1446467860

Following the instruction in doc/build.md I first built an approved version closest to the mentioned version on the issue posted. (USD Version 22.05b) using the default build options, nothing was added.
```
cd C:\dev\USD_builds\tags\v22.05b\USD\build_scripts
python build_usd.py "C:\dev\USD_builds\22.05b"
```

Then built the Maya-usd plugin (0.22.0) and was successful, see image:
![image](https://user-images.githubusercontent.com/88895104/221965362-f9cd5a42-e2e8-425c-9272-12729d7e9f30.png)

Here is the command used to build Maya-usd

```
cd C:\dev\maya-usd_builds\tags\v0.22.0\maya-usd
python build.py --maya-location "C:\Program Files\Autodesk\Maya2023" --pxrusd-location C:\dev\USD_builds\22.05b --devkit-location C:\dev\maya-usd_builds\devkit_2023.3\devkitBase --generator Ninja --build-args="-DBUILD_ADSK_PLUGIN=ON,-DBUILD_AL_PLUGIN=OFF,-DBUILD_PXR_PLUGIN=OFF,-DBUILD_TESTS=OFF,-DBUILD_WITH_PYTHON_3=ON,-DCMAKE_MAKE_PROGRAM='C:\dev\ninja\ninja.exe'" C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3
```

I followed the steps to load the plugin, I added the path to the MAYA_MODULE_PATH
No older versions of USD were in the PATH environment.
```
SET MAYA_MODULE_PATH=C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3\install\RelWithDebInfo

cd C:\Program Files\Autodesk\Maya2023\bin

start maya.exe
```

Unfortunately when I try to load the plugin I get an error "Unable to dynamically load"

```
// Error: file: C:/Program Files/Autodesk/Maya2023/scripts/others/pluginWin.mel line 316: Unable to dynamically load : C:/dev/maya-usd_builds/installed_plugins/maya-usd_2023.3/install/RelWithDebInfo/plugin/adsk/plugin/mayaUsdPlugin.mll
// The specified module could not be found.
// Error: file: C:/Program Files/Autodesk/Maya2023/scripts/others/pluginWin.mel line 316: The specified module could not be found.
//  (mayaUsdPlugin)
```

The mayaUsdPlugin.mll is there in the described path where it was built.

**Build log**
[build_log.txt](https://github.com/Autodesk/maya-usd/files/10854086/build_log.txt)

**Specs:**
 - OS Windows 10
 - Maya version 2023.3
 - Python 3.9.7
 - Maya-usd 0.22.0
 - USD Version 22.05b

**Additional context**
Bifrost is installed and its .mod file is adding a path:
```
packs/usd_pack/0.21.11/thirdparty/usd-0.21.11/bin
```

Not sure if this has anything to do with it but I tried removing it from the path and still could not dynamically load it.

I was hoping to see if anyone else had been successful in building the Maya-usd plugin with a later version of USD and if they would know anything thing I am missing.

Any insight would be very much appreciated.

Cheers,

Jason Coelho

**Comment by pierrebai-adsk** [comment]
Hi,

For the line `SET MAYA_MODULE_PATH=C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3\install\RelWithDebInfo`, I add MayaUSD at the end: `SET MAYA_MODULE_PATH=C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3\install\RelWithDebInfo\MayaUSD`

That may be all that is missing.

**Comment by jcoelho88** [comment]
> Hi,
> 
> For the line `SET MAYA_MODULE_PATH=C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3\install\RelWithDebInfo`, I add MayaUSD at the end: `SET MAYA_MODULE_PATH=C:\dev\maya-usd_builds\installed_plugins\maya-usd_2023.3\install\RelWithDebInfo\MayaUSD`
> 
> That may be all that is missing.

Hi @pierrebai-adsk,

Thanks for the response, For some reason there is no MayaUSD Folder within ".../RelWithDebInfo"

![image](https://user-images.githubusercontent.com/88895104/221966818-9698ad24-7b23-4356-bda4-731645a00c58.png)

Adding it anyway the plugin will no longer show up with this path in the plugin manager window.

Do you think this is a sign that the build its self is not completing correctly?

Cheers,

Jason Coelho

**Comment by pierrebai-adsk** [comment]
No, sorry, it must just be that I'm using a slightly different command line when building. Sorry for the noise!

**Comment by jcoelho88** [comment]
> No, sorry, it must just be that I'm using a slightly different command line when building. Sorry for the noise!

No worries, its still appreciated!

**Comment by pierrebai-adsk** [comment]
Here is the command-line I use to build the plugin, cut into multiple lines for legibility:

```
M:\Dev\Autodesk\Maya\builds\main\maya\build\RelWithDebInfo\runTime\bin\mayapy.exe"
"M:\Dev\Autodesk\Maya\worktrees\ecg-maya-usd\maya-usd\build.py"
--generator Ninja
--build-relwithdebug
--maya-location "M:\Dev\Autodesk\Maya\builds\main\maya\build\RelWithDebInfo\runTime"
--devkit-location "M:\Dev\Autodesk\Maya\builds\main\maya\build\RelWithDebInfo\runTime\..\Extra\devkitBase"
--pxrusd-location "M:\Dev\Autodesk\Maya\artifactory\Windows\USD\4ef2193"
--qt-location "M:\Dev\Autodesk\Maya\artifactory\Windows\qt\5.15.2\7a1c16f"
"--build-args=-DCMAKE_MAKE_PROGRAM=M:/Dev/Autodesk/Maya/artifactory/Windows/ninja/1.9.0/c68f192/ninja.exe,-DPYTHON_INCLUDE_DIR=M:/Dev/Autodesk/Maya/builds/main/maya/build/RelWithDebInfo/runTime/include/Python310/Python,-DPython_EXECUTABLE=M:/Dev/Autodesk/Maya/builds/main/maya/build/RelWithDebInfo/runTime/bin/mayapy.exe,-DPYTHON_LIBRARIES=M:/Dev/Autodesk/Maya/builds/main/maya/build/RelWithDebInfo/runTime/lib/python310.lib,-DBUILD_WITH_PYTHON_3=ON,-DBUILD_WITH_PYTHON_3_VERSION=3.10,-DCMAKE_WANT_MATERIALX_BUILD=ON,-DCMAKE_PREFIX_PATH=M:\Dev\Autodesk\Maya\artifactory\Windows\USD\4ef2193"
--stages=build
--redirect-outstream-file=False
--install-location M:\Dev\Autodesk\Maya\builds\ecg-maya-usd\install\RelWithDebInfo\MayaUSD
M:\Dev\Autodesk\Maya\builds\ecg-maya-usd
```

**Comment by pierrebai-adsk** [comment]
FWIW, when I had trouble with DLL dependencies on Windows, I enabled the OS DLL diagnostic mode. You need to then run Maya in a debugger and you will see enormous amount of information about each DLL that gets loaded and details why a given DLL fails to load. The command to enable this, which requires Windows SDK, is this:

```
"C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\gflags" -i Maya.exe +sls
```

To disable it afterwards:

```
"C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\gflags" -i Maya.exe -sls
```

**Comment by jcoelho88** [comment]
Hi @pierrebai-adsk,

Thanks for sharing, this is sure to shed some light on what's going on with my build, this is great!

Cheers,

Jason Coelho

**Comment by seando-adsk** [comment]
Inside of Maya can you run ```getenv "PATH"``` and ```getenv "PYTHONPATH"``` and make sure there are no other USD versions present (except for the one you built). It might be that you still have the MayaUsd plugin from the 2023 installer.

**Comment by jcoelho88** [comment]
> Inside of Maya can you run `getenv "PATH"` and `getenv "PYTHONPATH"` and make sure there are no other USD versions present (except for the one you built). It might be that you still have the MayaUsd plugin from the 2023 installer.

Hi @seando-adsk,

That's a good point, I'll check it out.

Cheers,

Jason Coelho

**Comment by jcoelho88** [comment]
Hi everyone,

Well it took me a while but I was able to successfully build the plugin with the help I got from this thread so thanks again for everyone who helped out.

USD was built again but this time I added the MaterialX and some other option flags like openvdb and opencolorio.

In the end I had to uninstall all the previous versions of the MayaUSD Extension from my computer (going back to 0.10.0)!

Specify the python build flags and edit a CMakeList - It was only supporting python 3.7 when I built it all with 3.9.

Here is an example of the problem this build was solving as mentioned in the first post of the issue, having instanced prims would crash Maya.


https://user-images.githubusercontent.com/88895104/222018535-fb5b3e61-fa40-4c43-b704-ab5b7060dfc9.mp4





And this is the results after loading the plugin built with Maya-usd 0.22.0 and USD 22.05b

https://user-images.githubusercontent.com/88895104/222018614-57a9232d-c0de-48b2-ad7d-97cc203fb199.mp4

Thanks again, I'll update the other ticket to let them know its a viable option.

Cheers,

Jason Coelho
